uqcsbot/scripts/wiki.py

```diff
@@ -0,0 +1,30 @@
+from uqcsbot import bot, Command
+import requests
+import json
+
+
+@bot.on_command("wiki")
+async def handle_wiki(command: Command):
+    search_query = command.arg
+    api_url = f"https://en.wikipedia.org/w/api.php?action=opensearch&format=json&limit=2"
+
+    http_response = await bot.run_async(requests.get, api_url, params={'search': search_query})
+    if http_response.status_code != requests.codes.ok:
+        bot.post_message(command.channel, "Problem fetching data")
+
+    _, title_list, snippet_list, url_list = json.loads(http_response.content)
+
+    # If the results are empty let them know. Any list being empty signifies this.
+    if len(title_list) == 0:
+        bot.post_message(command.channel, "No Results Found.")
+        return
+
+    title, snippet, url = title_list[0], snippet_list[0], url_list[0]
+    # Detect if there are multiple references to the query, if so, use the first reference (i.e. the second item in the lists).
+    multiple_reference_instances = ("may refer to:", "may have several meanings:")
+    if any(instance in snippet for instance in multiple_reference_instances) or not snippet:
```

Comment:
taking the second reference if snippet is empty doesn't really make sense, as this if condition is entirely about finding if the response contains multiple reference.

Can you provide a link to an example that produces an empty snippet? we can decide what the best action is from there I guess